src/everest/everest_storage.py

```diff
@@ -0,0 +1,922 @@
+            Path.mkdir(self._output_dir / "optimizer", parents=True)
+
+        return self._output_dir / "optimizer"
+
+    @property
+    def ensembles(self) -> dict[str, _OptimizerOnlyEnsemble]:
+        return {
+            str(d): _OptimizerOnlyEnsemble(self._output_dir / "ensembles" / d)
+            for d in os.listdir(self._output_dir / "ensembles")
+            if "batch_" in d
+        }
+
+    def get_ensemble_by_name(self, name: str) -> _OptimizerOnlyEnsemble:
+        if name not in self._ensembles:
+            self._ensembles[name] = _OptimizerOnlyEnsemble(
+                self._output_dir / "ensembles" / name
+            )
+
+        return self._ensembles[name]
+
+
+@dataclass
+class _EvaluationResults(TypedDict):
+    realization_controls: polars.DataFrame
+    batch_objectives: polars.DataFrame
+    realization_objectives: polars.DataFrame
+    batch_constraints: polars.DataFrame | None
+    realization_constraints: polars.DataFrame | None
+
+
+@dataclass
+class _GradientResults(TypedDict):
+    batch_objective_gradient: polars.DataFrame | None
+    perturbation_objectives: polars.DataFrame | None
+    batch_constraint_gradient: polars.DataFrame | None
+    perturbation_constraints: polars.DataFrame | None
+
+
+@dataclass
+class _MeritValue:
+    value: float
+    iter: int
+
+
+class EverestStorage:
+    def __init__(
+        self,
+        output_dir: Path,
+    ) -> None:
+        self._control_ensemble_id = 0
+        self._gradient_ensemble_id = 0
+
+        self._output_dir = output_dir
+        self._merit_file: Path | None = None
+        self.data = EverestStorageDataFrames()
+
+    @staticmethod
+    def _rename_ropt_df_columns(df: polars.DataFrame) -> polars.DataFrame:
+        """
+        Renames columns of a dataframe from ROPT to what will be displayed
+        to the user.
+        """
+        scaled_cols = [c for c in df.columns if c.lower().startswith("scaled")]
+        if len(scaled_cols) > 0:
+            raise ValueError("Scaled columns should not be stored into Everest storage")
+
+        renames = {
+            "objective": "objective_name",
+            "weighted_objective": "total_objective_value",
+            "variable": "control_name",
+            "variables": "control_value",
+            "objectives": "objective_value",
+            "constraints": "constraint_value",
+            "nonlinear_constraint": "constraint_name",
+            "perturbed_variables": "perturbed_control_value",
+            "perturbed_objectives": "perturbed_objective_value",
+            "perturbed_constraints": "perturbed_constraint_value",
+            "evaluation_ids": "simulation_id",
+        }
+        return df.rename({k: v for k, v in renames.items() if k in df.columns})
+
+    @staticmethod
+    def _enforce_dtypes(df: polars.DataFrame) -> polars.DataFrame:
+        dtypes = {
+            "batch_id": polars.UInt32,
+            "result_id": polars.UInt32,
```

Comment:
Could you drop all instances of storing `result_id`? I do not think it is needed anywhere and I am considering changing/dropping it in `ropt`, so it would be better not to depend on it.